bin/kill-tests

```diff
@@ -5,35 +5,61 @@
 
 require_relative '../deployment'
 require 'cdo/chat_client'
+require 'cdo/only_one'
 
-ChatClient.log "killing tests via #{$0}"
+SUCCESS_MSG = 'successfully killed all tests'.freeze
+FAIL_MSG = 'failed to kill all tests'.freeze
+
+# @return [Boolean] Returns whether tests are currently running via pgrep.
+def tests_running?
+  !`pgrep -f runner.rb`.empty? || !`pgrep -f cucumber`.empty? || !`pgrep -f 'rake test'`.empty?
+end
+
+# Prints to the console and logs to ChatClient a message.
+# @param message [String] The message to print and chat.
+def print_and_chat(message)
+  puts message
+  ChatClient.log message
+end
 
 def kill_tests
-  # kill dashboard unit tests
+  # Kill dashboard unit tests.
   `pkill -f 'rake test'`
 
-  # kill ui tests
+  # Kill ui tests.
   `pkill -f runner.rb`
   `pkill -f cucumber`
 end
 
-def tests_running?
-  !`pgrep -f runner.rb`.empty? || !`pgrep -f cucumber`.empty? || !`pgrep -f 'rake test'`.empty?
-end
+def main
+  unless [:adhoc, :development, :test].include? rack_env
+    puts "Cannot kill tests on #{rack_env}..."
+    return
+  end
+
+  unless tests_running?
+    puts "No tests are currently running on #{rack_env}..."
+    return
+  end
+
+  print "Kill tests on #{rack_env}? (Y/N): "
+  return unless gets.chomp.upcase == 'Y'
```

Comment:
It is, AFAICT. The `/restart-test-build` command in Slack POSTs to https://test.code.org/api/dev/start-build, which does not use this script.